hudi-client/src/main/java/org/apache/hudi/client/config/AbstractConfig.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.config;
+
+import java.util.Properties;
+
+/**
+ * Abstract config for multi-engine.
+ */
+public abstract class AbstractConfig<T> {
+  protected Properties props;
+
+  public AbstractConfig(Properties props) {
+    this.props = props;
+  }
+
+  public abstract long getMaxMemoryAllowedForMerge(String maxMemoryFraction);
+
+  public abstract T getWriteStatusStorageLevel();
+
+  public abstract T getBloomIndexInputStorageLevel();
+}
```

Comment:
Could only use `SparkConfig` to place all these methods if only consider spark free, not multi-engine

Reply:
I feel we can just make it Spark free for now.. Multi-engine can be dealt with separately..?